### PR TITLE
Fix Fetch stories based on translation

### LINKED ISSUE #101: Fetch Stories Based on Language

### PR DESCRIPTION
**Pull request by khaldoon9kh**
Add filter to fetched stories that change depending on the selected language.

### CODE REVIEW

File: src/containers/Stories/index.jsx [hunk 1]
```diff
@@ -12,10 +14,25 @@ const Stories = () => {
...
-      setPosts(fetchedPosts);
+      //filter posts based on the language
+      const filteredPosts = await fetchedPosts.filter((post) => {
+        return post.tags.toString() === langRef;
+      });
+      setPosts(filteredPosts);
     };
     fetchPosts();
-  }, []);
+  }, [langRef]);
```
Comment (halitbatur) [inline src/containers/Stories/index.jsx]: its better to do this for conditional rendering inorder not to cause alot of fetches whenever the language changes

File: src/containers/Stories/index.jsx [hunk 2]
```diff
@@ -2,8 +2,10 @@ import React from "react";
 import { Container } from "react-bootstrap";
 import SingleStoryCard from "../../components/SingleStoryCard";
 import LoadingPage from "../../components/LoadingPage";
+import i18next from "i18next";
 
 const Stories = () => {
+  const [langRef, setLangRef] = React.useState("10221");
```
Comment (halitbatur) [inline src/containers/Stories/index.jsx]: magic string

File: src/containers/Stories/index.jsx [hunk 3]
```diff
@@ -12,10 +14,25 @@ const Stories = () => {
...
     fetchPosts();
-  }, []);
+  }, [langRef]);
+
+  i18next.on("languageChanged", (lng) => {
+    let ref = i18next.translator.language;
+    if (ref === "en") {
+      setLangRef("10221");
+    } else if (ref === "ar") {
+      setLangRef("38299");
```
Comment (halitbatur) [inline src/containers/Stories/index.jsx]: magic strings please assign all these refs to a variable

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/containers/Stories/index.jsx (hunk 1 above)
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: I did not understand hao should I change...
Can you give me more details?

File: src/containers/Stories/index.jsx (hunk 2 above)
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: revised

File: src/containers/Stories/index.jsx (hunk 3 above)
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: revised

File: src/containers/Stories/index.jsx (hunk 1 above)
Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: fixed

**Comment by khaldoon9kh** [comment]
PTAL

File: src/containers/Stories/index.jsx [hunk 4]
```diff
@@ -2,8 +2,13 @@ import React from "react";
 import { Container } from "react-bootstrap";
 import SingleStoryCard from "../../components/SingleStoryCard";
 import LoadingPage from "../../components/LoadingPage";
+import i18next from "i18next";
 
 const Stories = () => {
+  const en = "10221";
```
Comment (halitbatur) [inline src/containers/Stories/index.jsx]: these consts should be all in caps since these won't be changing

**Review by halitbatur — CHANGES_REQUESTED** [review]

File: src/containers/Stories/index.jsx (hunk 4 above)
Comment (halitbatur) [reply src/containers/Stories/index.jsx]: and lets make it more specific smth like ENGLISH_LANGUAGE

Comment (khaldoon9kh) [reply src/containers/Stories/index.jsx]: Fixed

**Comment by khaldoon9kh** [comment]
PTAL

**Review by fadime-ozdemir — APPROVED** [review]
LGTM

**Review by halitbatur — APPROVED** [review]

**Pull request merged**